api/users.go

```diff
@@ -162,6 +199,108 @@ func (a *API) verifyUserAccountHandler(w http.ResponseWriter, r *http.Request) {
 	httpWriteJSON(w, res)
 }
 
+// userVerificationCodeInfoHandler handles the request to get the verification
+// code information of a user. It requires the user email to be provided. It
+// returns the user email, the verification code, the phone number, the code
+// expiration and if the code is valid (not expired and has not reached the
+// maximum number of attempts). If the user is already verified, an error is
+// returned. If the user is not found, an error is returned. If the
+// verification code is not found, an error is returned. If any other error
+// occurs, a generic error is returned.
+func (a *API) userVerificationCodeInfoHandler(w http.ResponseWriter, r *http.Request) {
+	// get the user email from the request query
+	userEmail := r.URL.Query().Get("email")
```

Comment:
@lucasmenendez is the sms part missing here?

Reply:
yes, it will be added in the next commit